From f62b3fdad70cc0e676fc2058660ba8cd708d0587 Mon Sep 17 00:00:00 2001
From: David Griffith <dave@661.org>
Date: Tue, 2 May 2017 19:06:18 -0700
Subject: [PATCH 43/68] 24-bit autocrop bug partial fix

This is from xv-20140327-mark-brader-24bit-autocrop-bug.msg.  It doesn't
entirely work, but is a decent first step.
---
 xvimage.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/xvimage.c b/xvimage.c
index eb48d80..ce99e6f 100644
--- a/xvimage.c
+++ b/xvimage.c
@@ -965,12 +965,12 @@ static int doAutoCrop24()
     for (i=0, misses=0, cp1=cp; i<cWIDE && misses<maxmiss; i++, cp1+=3) {
       r=cp1[0]-bgR;  g=cp1[1]-bgG;  b=cp1[2]-bgB;
       R=cp1[0]-oldr; G=cp1[1]-oldg; B=cp1[2]-oldb;
-      if (!inabsrange(r-g, EPSILON) ||
-	  !inabsrange(r-b, EPSILON) ||
-	  !inabsrange(b-g, EPSILON) ||
-	  !inabsrange(R-G, NEIGHBOR) ||
-	  !inabsrange(R-B, NEIGHBOR) ||
-	  !inabsrange(B-G, NEIGHBOR)) misses++;
+      if (!inabsrange(r, EPSILON) ||
+	  !inabsrange(r, EPSILON) ||
+	  !inabsrange(b, EPSILON) ||
+	  !inabsrange(R, NEIGHBOR) ||
+	  !inabsrange(R, NEIGHBOR) ||
+	  !inabsrange(B, NEIGHBOR)) misses++;
       oldr=r; oldg=g; oldb=b;
     }
 
-- 
2.17.1

